### PR TITLE
change kubelet probe metrics to counter

### LINKED ISSUE #75839: kubelet 'metrics/probes' endpoint uses a gauge instead of counters
Currently the kubelet exports liveness/readiness probe metric information at an [endpoint 'metrics/probes'](https://github.com/kubernetes/kubernetes/blob/5c2fe63591acab508e1ff5771c3aa70268852318/pkg/kubelet/server/server.go#L79). 

The metric data looks like this:

```
# HELP prober_probe_result The result of a liveness or readiness probe for a container.
# TYPE prober_probe_result gauge
prober_probe_result{container="etcd-container",container_name="etcd-container",namespace="kube-system",pod="etcd-server-events-kubernetes-master",pod_name="etcd-server-events-kubernetes-master",pod_uid="",probe_type="Liveness"} 0
prober_probe_result{container="etcd-container",container_name="etcd-container",namespace="kube-system",pod="etcd-server-kubernetes-master",pod_name="etcd-server-kubernetes-master",pod_uid="",probe_type="Liveness"} 0
prober_probe_result{container="fluentd-gcp",container_name="fluentd-gcp",namespace="kube-system",pod="fluentd-gcp-v3.2.0-g7ftb",pod_name="fluentd-gcp-v3.2.0-g7ftb",pod_uid="",probe_type="Liveness"} 0
prober_probe_result{container="kube-apiserver",container_name="kube-apiserver",namespace="kube-system",pod="kube-apiserver-kubernetes-master",pod_name="kube-apiserver-kubernetes-master",pod_uid="",probe_type="Liveness"} 0
prober_probe_result{container="kube-apiserver",container_name="kube-apiserver",namespace="kube-system",pod="kube-apiserver-kubernetes-master",pod_name="kube-apiserver-kubernetes-master",pod_uid="",probe_type="Readiness"} 0
prober_probe_result{container="kube-controller-manager",container_name="kube-controller-manager",namespace="kube-system",pod="kube-controller-manager-kubernetes-master",pod_name="kube-controller-manager-kubernetes-master",pod_uid="",probe_type="Liveness"} 0
prober_probe_result{container="kube-scheduler",container_name="kube-scheduler",namespace="kube-system",pod="kube-scheduler-kubernetes-master",pod_name="kube-scheduler-kubernetes-master",pod_uid="",probe_type="Liveness"} 0
prober_probe_result{container="l7-lb-controller",container_name="l7-lb-controller",namespace="kube-system",pod="l7-lb-controller-v1.2.3-kubernetes-master",pod_name="l7-lb-controller-v1.2.3-kubernetes-master",pod_uid="",probe_type="Liveness"} 0
```

This is incredibly useful metric information, however, it would be better if these metrics were counters rather than gauges. Currently a success probe is registered as a '0' value and a failure is registered as a '1' value. However, depending on your prometheus scrape interval, you can lose bits of information. It would be better if we emitted these metrics as counters.

/sig instrumentation node

### PR DESCRIPTION
**Pull request by danielqsj**
**What type of PR is this?**

/kind feature
/sig instrumentation node

**What this PR does / why we need it**:

As discussion in https://github.com/kubernetes/kubernetes/issues/75839, we prefer to using counter type of metrics for kubelet probe rather than gauge type.

**Which issue(s) this PR fixes**:

Fixes #75839

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Change kubelet probe metrics to counter type.
The metrics `prober_probe_result` is replaced by `prober_probe_total`.
```

### CODE REVIEW

**Comment by brancz** [comment]
Given that none of the popular alert/dashboard definitions use this metric, I'm inclined to just remove the old metric.

File: pkg/kubelet/prober/worker.go [hunk 1]
```diff
@@ -98,16 +101,27 @@ func newWorker(
 		w.initialValue = results.Success
 	}
 
-	w.proberResultsMetricLabels = prometheus.Labels{
-		"probe_type":     w.probeType.String(),
-		"container_name": w.container.Name,
```
Comment (logicalhan) [inline pkg/kubelet/prober/worker.go]: Woah, this is weird. I realize you didn't write this, but curious if anyone knows why we would have multiple labels on a metric which would always share the same label value?

**Review by logicalhan — COMMENTED** [review]
I'm okay with just deleting the old metric as well. As mentioned in the issue, I would be surprised if anyone is actually using it for alerting since it is probably pretty noisy.

File: pkg/kubelet/prober/worker.go (hunk 1 above)
Comment (brancz) [reply pkg/kubelet/prober/worker.go]: Are you talking about container and contaiber_name?

Comment (logicalhan) [reply pkg/kubelet/prober/worker.go]: Either. The previous implementation of this seems to store the same value for in both the container label and the container_name label. Same thing happens for pod and pod_name, which I find quite odd.

Comment (logicalhan) [reply pkg/kubelet/prober/worker.go]: You can see it when you examine the actual metric output via curl. 

```# TYPE prober_probe_result gauge
prober_probe_result{container="etcd-container",container_name="etcd-container",namespace="kube-system",pod="etcd-server-events-kubernetes-master",pod_name="etcd-server-events-kubernetes-master",pod_uid="1234",probe_type="Liveness"} 0```

Why do we have two labels which always have the same value in either of them?

Comment (logicalhan) [reply pkg/kubelet/prober/worker.go]: Super weird.

Comment (brancz) [reply pkg/kubelet/prober/worker.go]: This was for migration purposes. For a long time these and cadvisor metrics used pod_name and container_name label keys, which violate the instrumentation guidelines. For migration purposes in 1.14 both are present.

Comment (logicalhan) [reply pkg/kubelet/prober/worker.go]: Ah, that would explain it.

**Comment by danielqsj** [comment]
@brancz @logicalhan removed old metrics `prober_probe_result`. PTAL

**Comment by brancz** [comment]
looks good from instrumentation side

/lgtm

still needs a kubelet approver though @derekwaynecarr @tallclair

**Review by logicalhan — COMMENTED** [review]
/lgtm

@dashpole, thoughts?

**Comment by dashpole** [comment]
/lgtm
This looks reasonable to me.

**Comment by danielqsj** [comment]
friendly ping @derekwaynecarr @tallclair , could you please help review it?

**Pull request merged**